### PR TITLE
Reincorporate local customizations

### PR DESCRIPTION
**Pull request by mistydemeo**
These two commits contain a few pieces of custom behaviour that we use in our hubot-slack. This PR is on top of #12, which brings us up to date with the latest hubot-slack.

### CODE REVIEW

File: src/client.coffee [hunk 1]
```diff
@@ -166,20 +168,32 @@ class SlackClient
 
     @robot.logger.debug "SlackClient#send() room: #{room}, message: #{message}"
 
+    text = if typeof message is 'string' then message else message.text
+    text = text.substring(0, SlackClient.MAX_MESSAGE_LENGTH)
```
Comment (wuest) [inline src/client.coffee]: Should we warn when truncating?  Or perhaps set `MAX_MESSAGE_LENGTH` to 7997 and append `'...'`?

Comment (mistydemeo) [reply src/client.coffee]: Good call, I'll do that.

**Comment by mistydemeo** [comment]
> Support clipping text content in messages to a max length (i think a reasonable enhancement upstream)

I can look at opening a PR for this. The value chosen here is the max value from the Slack API.

> Manually "unfurling" certain types of links - this one confuses me, since this could be implemented using the response middleware chain, right?

It probably could, yes. We did it here just because we're sending as an attachment, which Slack won't unfurl.

> Opting out of ever having simple messages, always depending on an attachment - this also confuses me and i'm not sure its intentional.

This is intentional! The attachment formatting has become a very useful visual shorthand to indicate that a bot is speaking, which is very important for us because Hubot frequently posts to Slack - in some rooms, more frequently than humans do.

> it seems to me that the change on line 196 is actually an incorrect use of the API for @slack/client's WebClient in v3.x - the text argument should either be a string or undefined, an empty object is not meaningful.

This was a change I made in this PR; before this it was an empty string in both cases. I'll switch back.

**Comment by aoberoi** [comment]
> I can look at opening a PR for this. The value chosen here is the max value from the Slack API.

Would be happy to review and land it 😄. I don't need a commitment on when, so at your own leisure is fine.

> This is intentional! The attachment formatting has become a very useful visual shorthand to indicate that a bot is speaking, which is very important for us because Hubot frequently posts to Slack - in some rooms, more frequently than humans do.

This is totally valid and very interesting. Thank you for sharing. I'm sure our designers would hope that the `APP` bubble next to the avatar and name would be enough of a visual indication, but you've creatively found a way to turn it up a notch. As for reducing redundancy (one of the reasons I wanted to pop in), I'm personally a little afraid of what this might break, so I don't think this kind of change is right for the `hubot-slack` project; but you should keep on doing it if it works for your team. This also might be doable using the response middleware chain.

**Comment by mistydemeo** [comment]
> I'm sure our designers would hope that the APP bubble next to the avatar and name would be enough of a visual indication, but you've creatively found a way to turn it up a notch.

This actually reminds me of one other reason that we do this, which I neglected to mention earlier. Long attachments are automatically truncated and are unfurlable; long messages are always printed in their entirety. Because our Hubot can post long responses, it's helpful to us to have those posted as Slack attachments so that they take up less screen space by default.

**Pull request merged**